diff --git a/node_modules/@dhis2/app-shell/index.html b/node_modules/@dhis2/app-shell/index.html
index dd10d7d..30a1ff1 100644
--- a/node_modules/@dhis2/app-shell/index.html
+++ b/node_modules/@dhis2/app-shell/index.html
@@ -10,6 +10,7 @@
 
         <!-- Injection point for base URL from backend -->
         <meta name="dhis2-base-url" content="__DHIS2_BASE_URL__" />
+        <meta name="dhis2-app-root-url" content="__DHIS2_APP_ROOT_URL__" />
 
         <!--
             manifest.json provides metadata used when your web app is installed on a
diff --git a/node_modules/@dhis2/app-shell/src/index.jsx b/node_modules/@dhis2/app-shell/src/index.jsx
index a2573cc..0ab2b3c 100644
--- a/node_modules/@dhis2/app-shell/src/index.jsx
+++ b/node_modules/@dhis2/app-shell/src/index.jsx
@@ -5,6 +5,18 @@ import App from './App.jsx'
 import 'typeface-roboto'
 import './index.css'
 
+const appRootUrl = document
+    .querySelector('meta[name="dhis2-app-root-url"]')
+    ?.getAttribute('content')
+if (appRootUrl && appRootUrl !== '__DHIS2_APP_ROOT_URL__') {
+    const head = document.querySelector('head')
+    const base = document.createElement('base')
+    // `href` needs ending slash to work:
+    base.setAttribute('href', appRootUrl + '/')
+    // Add to end of head, to avoid interfering with loaded assets
+    head.appendChild(base)
+}
+
 const container = document.getElementById('dhis2-app-root')
 const root = createRoot(container)
 root.render(
